File: web/src/lib/tags.test.ts
import { describe, expect, it } from 'vitest';
import { addTag, normalizeTag, tagColorIndex } from './tags.ts';

describe('card tags', () => {
  it('etiketi normalize eder ve Türkçe büyük-küçük harf tekrarını engeller', () => {
    expect(normalizeTag('  Kolay   İş ')).toBe('Kolay İş');
    expect(addTag(['İş'], 'iş').error).toBe('Bu etiket zaten ekli.');
    expect(addTag(['Backend API'], 'backend api').error).toBe('Bu etiket zaten ekli.');
    expect(addTag(['Backend'], ' API ').tags).toEqual(['Backend', 'API']);
  });

  it('uzunluk ve adet sınırını uygular, rengi kararlı üretir', () => {
    expect(addTag([], 'x'.repeat(31)).error).toContain('30');
    expect(addTag(Array.from({ length: 10 }, (_, i) => `tag-${i}`), 'fazla').error).toContain('10');
    expect(tagColorIndex('Backend')).toBe(tagColorIndex('Backend'));
    expect(tagColorIndex('Backend')).toBeGreaterThanOrEqual(0);
    expect(tagColorIndex('Backend')).toBeLessThan(8);
  });
});

File: mobile/test/tags_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:planner/tags.dart';

void main() {
  test('etiketi temizler ve büyük-küçük harf tekrarını engeller', () {
    final added = addCardTag(const [], '  Backend   API  ');
    expect(added.error, isNull);
    expect(added.tags, ['Backend API']);

    final duplicate = addCardTag(added.tags, 'backend api');
    expect(duplicate.tags, added.tags);
    expect(duplicate.error, isNotNull);

    expect(addCardTag(const ['İş'], 'iş').error, isNotNull);
  });

  test('etiket uzunluğu ve kart başına etiket sayısı sınırlıdır', () {
    final long = addCardTag(
      const [],
      List.filled(maxTagLength + 1, 'x').join(),
    );
    expect(long.error, isNotNull);

    final full = List.generate(maxCardTags, (index) => 'tag-$index');
    final overflow = addCardTag(full, 'extra');
    expect(overflow.tags, full);
    expect(overflow.error, isNotNull);
  });

  test('etiket rengi aynı metin için kararlıdır', () {
    expect(tagColorIndex('Backend'), tagColorIndex('Backend'));
    expect(tagColorIndex('Backend'), inInclusiveRange(0, 7));
  });
}
